godtools/App/Features/Dashboard/Presentation/Tools/Presentation/Subviews/ToolFilterSelectionView/ViewModel/ToolFilterSelectionViewModel.swift

```diff
@@ -0,0 +1,43 @@
+//
+//  ToolFilterSelectionViewModel.swift
+//  godtools
+//
+//  Created by Rachael Skeath on 8/25/23.
+//  Copyright © 2023 Cru. All rights reserved.
+//
+
+import Foundation
+import Combine
+
+class ToolFilterSelectionViewModel: ObservableObject {
+    
+    var selectedCategory: ToolCategoryDomainModel? = nil
```

Comment:
Should these domain model references be immutable or at least private?  Unless they will need to be mutated by the subclasses?

Reply:
I ended up removing these in favor of a `ToolFilterSelection` current value publisher in my follow-up branch.  But yes, that's a huge downside of using inheritance when we really just want a protocol-- we end up with a lot of public stuff.  If there's a better way to reuse views with different viewModel types, I'm all for it, just not sure what it is 🤔

Reply:
That makes sense, I hadn't thought of having the interfaces injected into a single viewModel.  I imagine that could still be tricky because there's likely specific behavioral differences that could be difficult to model in an interface.  I'll keep going as is, but it wouldn't be hard to make two separate views if that ends up making more sense.  I know we've had similar conversations to this before, I think I'm so used to protocols that I really want the baseViewModel thing to work 😅

Reply:
Yeah interfaces are the preferred way.  And it can get a bit tricky too with interfaces.  But yeah keep at it, will be interesting to see how it goes.